state-sync/storage-service/types/src/lib.rs

```diff
@@ -495,14 +498,12 @@ impl<T: PrimInt> CompleteDataRange<T> {
     }
 
     /// Returns the length of the data range.
-    ///
-    /// Note that `CompleteDataRange` maintains the invariant that the range
-    /// length is always expressible without overflow, so this method is
-    /// infallible.
     #[inline]
-    pub fn len(&self) -> T {
-        debug_assert!(range_length_checked(self.lowest, self.highest).is_ok());
-        (self.highest - self.lowest) + T::one()
+    pub fn len(&self) -> Result<T, DegenerateRangeError> {
+        self.highest
```

Comment:
Aah, I see what you mean here. Let's leave this as-is, I think it's fine 😄

Reply:
I assume that "as-is" is to go with my current change?

Reply:
Yeah, you can leave it like it is here (after your change) 😄